src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java

```diff
@@ -456,6 +455,18 @@ void updateDataset(@Bind("datasetId") Integer datasetId,
             " WHERE d.dac_id IN (<dacIds>) ")
     Set<DatasetDTO> findDatasetsByDacIds(@BindList("dacIds") List<Integer> dacIds);
 
+    @UseRowReducer(DatasetReducer.class)
+    @SqlQuery("""
+            SELECT distinct d.*, k.key, p.property_value, c.consent_id, d.dac_id, c.translated_use_restriction
+            FROM dataset d
+            LEFT JOIN dataset_property p ON p.dataset_id = d.dataset_id
+            LEFT JOIN dictionary k ON k.key_id = p.property_key
+            LEFT JOIN consent_associations a ON a.dataset_id = d.dataset_id
+            LEFT JOIN consents c ON c.consent_id = a.consent_id
+            WHERE d.dac_id IN (<dacIds>)
+        """)
+    List<Dataset> findDatasetListByDacIds(@BindList("dacIds") List<Integer> dacIds);
```

Comment:
I just want to say I really like this dao pattern

Reply:
I'm enjoying this too - it's so much easier to read as well as test if you have a db app up and running.